pkg/hive/hive.go

```diff
@@ -71,80 +67,73 @@ func (s *Service) Protocol() p2p.ProtocolSpec {
 // Init is called when the new peer is being initialized.
 // This should happen after overlay handshake is finished.
 func (s *Service) Init(ctx context.Context, peer p2p.Peer) error {
-	for i := 0; i < maxPO; i++ {
-		// todo: figure out the limit
-		resp, err := s.requestPeers(ctx, peer, i, 10)
-		if err != nil {
-			return err
-		}
-
-		for _, newPeer := range resp {
-			addr, err := ma.NewMultiaddr(newPeer)
-			if err != nil {
-				s.logger.Infof("Connect failed for %s: %w", newPeer, err)
-				continue
-			}
-
-			if _, err := s.connecter.Connect(ctx, addr); err != nil {
-				s.logger.Infof("Connect failed for %s: %w", addr.String(), err)
+	go func() {
```

Comment:
consider the following for an even further simplification (where even this `Init` would not be needed at all):
1. When topology driver broadcasts every connected peer _to_ a newly added peer
2. When topology driver broadcasts every newly connected peer _to_ every connected peer (everyone else)

Reply:
i'd like to add something that came into mind, is that we _might_ need a message exchange to ask for peers, because if we go with an approach of just normal gossiping - we'd indeed need to keep a queue of peers we'd be able to connect to when saturation depth decreases (but this is a kademlia implementation detail. it might not infer a message exchange at all)

Reply:
Makes sense, let's ee when it comes, and we can add it. There is no problem adding another stream for fetching the peers to a protocol, and it will work, along this "broadcast" style of receiving peers, no problem.